src/domain/VehicleFactory.ts

```diff
@@ -33,6 +34,8 @@ export class VehicleFactory {
         return new processors.LgvProcessor(vehicleObj as LightGoodsVehicle, techRecordDAO);
       case enums.VEHICLE_TYPE.CAR:
         return new processors.CarProcessor(vehicleObj as Car, techRecordDAO);
+      case enums.VEHICLE_TYPE.SMALL_TRL:
+        return new processors.SmallTrailerProcessor(vehicleObj as SmallTrailer, techRecordDAO);
```

Comment:
I've been trying to rack my brains for a clean way we can remove the type casting `vehicleObj as ...` but I think unless we use chained if statements that utilise type guards, there's no other way....

```
function isSmallTrl(vehicleObj: Vehicle): vehicleObj is SmallTrailer {
    return vehicleObj.techRecord[0].vehicleType === enums.VEHICLE_TYPE.SMALL_TRL
}

if (isSmallTrl(vehicleObj)) {
  return new processors.SmallTrailerProcessor(vehicleObj, techRecordDAO);
}
else if {...}
```

Reply:
I know typecasting is not the best approach but your suggestion introduces 7 new methods just to be used in `if else` statements. Thus, I will leave it as is.

Maybe one day we will have proper interfaces for each vehicle type. As that is the proper fix to the root problem here.